### PR TITLE
Story cadp 20561 - DB restoration - VKM and CAKM

### PR DESCRIPTION
**Pull request by t0262653**

### CODE REVIEW

File: cakm/ekm/cakm_ekm_db_restoration.sql [hunk 1]
```diff
@@ -0,0 +1,154 @@
...
+CREATION_DISPOSITION = OPEN_EXISTING;';
+
+--PRINT @sql;
+EXEC sp_executesql @sql;
+--SELECT name,thumbprint FROM SYS.ASYMMETRIC_KEYS where name = @keyname;
+
+PRINT "Setting login and cred for the key"
+
+SET @login_cmd = '
+CREATE LOGIN ' + QUOTENAME(@login) + '
```
Comment (SheffaliG) [inline cakm/ekm/cakm_ekm_db_restoration.sql:93]: What if these queries fail ?

File: cakm/ekm/cakm_ekm_db_restoration.sql [hunk 2]
```diff
@@ -0,0 +1,154 @@
...
+SET @drop_sql = '
+drop ASYMMETRIC KEY ' + QUOTENAME(@keyname) + ';';
+
+BEGIN TRY
+	PRINT 'Restoring DB';
+	RESTORE DATABASE @dbname FROM DISK = @backupp WITH REPLACE;
+    PRINT 'Restore successful.';
+END TRY
+BEGIN CATCH
+    PRINT 'Restore failed. Error: ' + CONVERT(VARCHAR(500), ERROR_MESSAGE());
```
Comment (SheffaliG) [inline cakm/ekm/cakm_ekm_db_restoration.sql:124]: Do we need to check the error received or only error condition is using UUID instead of fingerprint?

Comment (t0262653) [reply cakm/ekm/cakm_ekm_db_restoration.sql:124]: This is the case when DB restoration is failed either with UUID or FP. i am not able to completely understand this comment.

File: cakm/ekm/cakm_ekm_db_restoration.sql (hunk 1 above)
Comment (t0262653) [reply cakm/ekm/cakm_ekm_db_restoration.sql:93]: error condition would be displayed on screen but script will continue.

**Review by gsharma-ad — APPROVED** [review]

**Pull request merged**